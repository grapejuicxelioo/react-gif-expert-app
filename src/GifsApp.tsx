import { mockGifs } from './mock-data/gifs.mock'
import { CustomHeader } from './shared/components/CustomHeader'
import { SearchBar } from './shared/components/SearchBar'
import { PreviousSearches } from './gifs/components/PreviousSearches'
import { GifList } from './gifs/components/GifList'
import { useState } from 'react'

export function GifsApp() {
    const [previousTerms, setPreviousTerms] = useState(['']);

    const handleTermClicked = (term: string) => {
        console.log({ term })
    };
    const handleSearch = (query: string = '') => {
        query = query.trim().toLowerCase();
        if (query.length === 0) return;

        if (previousTerms.includes(query)) return;

        // const currentTerms = previousTerms.slice(0, 8);
        // currentTerms.unshift(query);

        // setPreviousTerms(currentTerms);

        setPreviousTerms([query, ...previousTerms].splice(0, 8));
    };

    return (
        <>
            {/* header */}
            <CustomHeader title='buscador de gifs' description='descubre y comparte el Gif perfecto' />

            {/* search */}
            <SearchBar placeHolder="buscar" onQuery={handleSearch} />

            {/* busquedas previas */}
            <PreviousSearches
                searches={previousTerms}
                onLabelClicked={handleTermClicked}
            />

            {/* gifs */}
            <GifList gifs={mockGifs} />
        </>
    )
}