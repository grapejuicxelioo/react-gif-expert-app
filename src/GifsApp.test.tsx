import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import { GifsApp } from './GifsApp'

describe("GifsApp", () => {
    test("should render ...", () => {
        const { container } = render(<GifsApp />);
        expect(container).toMatchSnapshot();
    })
})