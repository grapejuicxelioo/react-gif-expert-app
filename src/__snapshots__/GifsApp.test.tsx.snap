// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > should render ... 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      buscador de gifs
    </h1>
    <p>
      descubre y comparte el Gif perfecto
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      placeholder="buscar"
      type="text"
      value=""
    />
    <button>
      buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      busquedas previas
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
